packit_service/worker/jobs.py

```diff
@@ -420,109 +470,100 @@ def process_jobs(self, event: Event) -> List[TaskResults]:
-                            success=False,
-                            msg="Account is not allowlisted!",
-                            job_config=job_config,
-                            event=event,
-                        )
-                    )
-                return processing_results
-
-            signatures = []
-
-            # we want to run handlers for all possible jobs, not just the first one
-            for job_config in job_configs:
-                if self.service_config.deployment not in job_config.packit_instances:
-                    logger.debug(
-                        f"Current deployment ({self.service_config.deployment}) "
-                        f"does not match the job configuration ({job_config.packit_instances}). "
-                        "The job will not be run."
-                    )
-                    continue
-
-                handler = handler_kls(
-                    package_config=event.package_config,
-                    job_config=job_config,
-                    event=event.get_dict(),
-                )
-                if not handler.pre_check():
-                    continue
-
-                if event.actor and not handler.check_if_actor_can_run_job_and_report(
-                    actor=event.actor
-                ):
-                    # For external contributors, we need to be more careful when running jobs.
-                    # This is a handler-specific permission check
-                    # for a user who trigger the action on a PR.
-                    # e.g. We don't allow using internal TF for external contributors.
-                    continue
-
-                self.report_task_accepted(
-                    event=event, handler=handler, job_config=job_config
-                )
-
-                signatures.append(
-                    handler_kls.get_signature(event=event, job=job_config)
-                )
-                processing_results.append(
+                return [
                     TaskResults.create_from(
-                        success=True,
-                        msg="Job created.",
+                        success=False,
+                        msg="Account is not allowlisted!",
                         job_config=job_config,
                         event=event,
                     )
-                )
-            # https://docs.celeryproject.org/en/stable/userguide/canvas.html#groups
-            group(signatures).apply_async()
+                    for job_config in job_configs
+                ]
+
+            processing_results.extend(
+                self.create_tasks(event, job_configs, handler_kls)
+            )
 
         return processing_results
 
-    def process_message(
-        self, event: dict, topic: str = None, source: str = None
+    def create_tasks(
+        self, event: Event, job_configs: List[JobConfig], handler_kls: Type[JobHandler]
     ) -> List[TaskResults]:
         """
```

Comment:
I decomposed the create_task, PTAL if you think it should be improved even more.

Reply:
@lbarcziova Everything looks really good, but I am still inclined to have `create_task` instead of `create_tasks`.

Otherwise, it's really a huge improvement and the code is now much easier to read.